File: src/main/java/play/pluv/login/application/dto/LoginTypeResponse.java
package play.pluv.login.application.dto;

import static play.pluv.playlist.domain.MusicStreaming.YOUTUBE;

import java.util.List;
import play.pluv.playlist.domain.MusicStreaming;

public record LoginTypeResponse(
    String type
) {

  public static LoginTypeResponse from(final MusicStreaming type) {
    if (type == YOUTUBE) {
      return new LoginTypeResponse("google");
    }
    return new LoginTypeResponse(type.getName());
  }

  public static List<LoginTypeResponse> createList(final List<MusicStreaming> types) {
    return types.stream()
        .map(LoginTypeResponse::from)
        .toList();
  }
}

File: src/main/kotlin/play/pluv/login/application/dto/LoginTypeResponse.kt
package play.pluv.login.application.dto

import play.pluv.playlist.domain.MusicStreaming

data class LoginTypeResponse(
    val type: String
) {
    companion object {
        fun from(type: MusicStreaming): LoginTypeResponse {
            if (type == MusicStreaming.YOUTUBE) {
                return LoginTypeResponse("google")
            }
            return LoginTypeResponse(type.name)
        }

        fun createList(types: List<MusicStreaming>): List<LoginTypeResponse> {
            return types
                .map { type: MusicStreaming -> from(type) }
                .toList()
        }
    }
}
